app/blueprints/api.py

```diff
@@ -28,8 +29,12 @@ def entry_list():
 
         if request.args.get('first_letter'):
             entries = entries.where('name_first_letter', '==', request.args.get('first_letter').lower())
-
+    
         results = [entry.to_dict() for entry in entries.get()]
+
+        if request.args.get('random'):
+            results = random.sample(results, int(request.args.get('random')))
```

Comment:
@kierana as discussed random will need cache-busting (based upon time?)

Reply:
I do this frontend - As the cache key is a URL, adding a timestamp to the URL breaks out the cache